presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeExpressionVisitor.java

```diff
@@ -119,8 +119,7 @@ public BytecodeNode visitCall(CallExpression call, final Scope scope)
                     generator = new RowConstructorCodeGenerator();
                     break;
                 case BIND:
-                    generator = new BindCodeGenerator();
-                    break;
+                    throw new UnsupportedOperationException();
```

Comment:
If I understand correctly, you moved `bind` into `FunctionCallCodeGenerator` for two reasons.

1. You want to know what the return type of bind need to be because it's in the function's lambdaInterface field
2. You want to avoid having to add "bind" call when you don't have anything to bind.

I don't think this is good enough reason to sacrifice the way `Bind` code is generated here.